app/src/androidTestEreferrals/java/org/eyeseetea/malariacare/data/sync/exporter/WSPushControllerShould.java

```diff
@@ -0,0 +1,112 @@
+import org.eyeseetea.malariacare.utils.Constants;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+public class WSPushControllerShould {
+
+    private static final String PUSH_RESPONSE_CONFLICT = "push_response_conflict.json";
+
+    private MockWebServer server;
+    private eReferralsAPIClient apiClient;
+    private WSPushController mWSPushController;
+    private String[] eventUIDs = {"LRR4ZZidQ6T", "PHp2WANFHE1", "NDqaWw51WJr", "Ian8YUgm7T3"};
+    private List<Long> surveysIDs = new ArrayList<>();
+
+
+    @Before
+    public void setUp() throws Exception {
+        this.server = new MockWebServer();
+        this.server.start();
+        apiClient = initializeApiClient();
+        mWSPushController = new WSPushController(apiClient);
+    }
+
+
+    @Test
+    public void put_survey_status_conflict_if_failed_or_no_success_action_status()
+            throws IOException {
+        initSurveys();
+        enqueueResponse(PUSH_RESPONSE_CONFLICT);
+        mWSPushController.push(new IPushController.IPushControllerCallback() {
+            @Override
+            public void onComplete() {
+                SurveyDB firstNoConflict = SurveyDB.findById(surveysIDs.get(0));
+                assertThat(firstNoConflict.getStatus(), is(Constants.SURVEY_SENT));
+                SurveyDB firstConflict = SurveyDB.findById(surveysIDs.get(1));
+                assertThat(firstConflict.getStatus(), is(Constants.SURVEY_CONFLICT));
+                SurveyDB secondConflict = SurveyDB.findById(surveysIDs.get(2));
+                assertThat(secondConflict.getStatus(), is(Constants.SURVEY_CONFLICT));
+                SurveyDB secondNoConflict = SurveyDB.findById(surveysIDs.get(3));
+                assertThat(secondNoConflict.getStatus(), is(Constants.SURVEY_SENT));
+            }
+
+            @Override
+            public void onInformativeError(Throwable throwable) {
+                boolean hasError = throwable != null;
+                assertThat(hasError, is(true));
+            }
+
+            @Override
+            public void onInformativeMessage(String message) {
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+                boolean hasError = throwable != null;
+                assertThat(hasError, is(false));
+            }
+        });
+    }
+
+    private void initSurveys() {
+        for (String eventUID : eventUIDs) {
+            SurveyDB surveyDB = new SurveyDB(new OrgUnitDB("test"), new ProgramDB("test"),
+                    new UserDB("test", "test"));
+            surveyDB.setStatus(Constants.SURVEY_COMPLETED);
+            surveyDB.setEventUid(eventUID);
+            surveyDB.save();
+            surveysIDs.add(surveyDB.getId_survey());
+        }
+    }
+
```

Comment:
@manuelplazaspalacio if you execute the test and then try to open the app, the app will crash. Ideally the test should work completely in memory, and not using the app's DB, otherwise subsequent tests will depend on the exit state of the previous ones, or in the worst scenarios, things like this can happen. Can you please change it either to implement the use of an in-memory DB for the tests (the best) either to reset the state of the DB at the end of the test? If you go for the 2nd option because of the lack of time, please remember to create an issue to refactor that in the future